### PR TITLE
Updated fish shell recipe

### PR DESCRIPTION
**Pull request by miqlas**
The configure.ac patch doesn't works and the networking code disabled almost completely, but it builds, packaging works too, and i was able to run it.

szlvlaj

If you have time, please help to fix it.

### CODE REVIEW

**Comment by miqlas** [comment]
I don't really know, whats wrong. Can somebody help me?

**Comment by korli** [comment]
Regarding configure.ac, please see http://www.clearchain.com/blog/posts/autotools for how to check for the host OS. You might as well want to just use "autoreconf -fi" which invokes autoconf, libtoolize when needed.

**Comment by miqlas** [comment]
OMG, it's like rocket techology for me.
Sorry guys.

**Comment by miqlas** [comment]
Thanks korli for the line endings hint. I converted it to LF, it should be good now.
I'll look into the autoconf thing.

**Comment by miqlas** [comment]
Thank you, PulkoMandy, your help is really appreciated.

**Comment by miqlas** [comment]
I changed it so:

-if test `uname` != "Darwin"; then
-   LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
-fi
+case $host_os in
+	darwin* )
+		;;
+	haiku*)
+		;;
+	*)
+		LDFLAGS_FISH="$LDFLAGS_FISH -rdynamic"
+		;;
+esac

But it somehow picks up the -rdynamic flag. I'm not sure, why. Maybe the $host_os is not set yet.

 Can somebody help me?

**Comment by miqlas** [comment]
The get-MAC-address part of the code disabled yet, because i can't find any solution for that. Can somebody look into that? Thanks!

**Comment by mmuman** [comment]
you messed up the case for the uname test, Darwin, not darwin, and Haiku, not haiku.

**Comment by miqlas** [comment]
It is working now correctly. Do you see any problem in the latest commit?

**Comment by miqlas** [comment]
Should i write:
darwin* | haiku* )?

**Comment by mmuman** [comment]
Oh ok with host_os instead it should work as lowercase.
No it's better to leave both cases distinct I think.

**Comment by mmuman** [comment]
btw 200015 sounds a bit far in the future for copyright term :D

**Comment by miqlas** [comment]
Fixed.

**Comment by miqlas** [comment]
Maybe i need define CFLAGS=-D_BSD_SOURCE for networking... Let me check at weekend...

**Comment by miqlas** [comment]
It is less hacky now, built correctly with the minimized patchset.
The fish_config doesn't works, i got the following error:

"socket: Address family not supported by protocol family"

Maybe it is a bug on Haiku side?

**Comment by miqlas** [comment]
Yeah, it is the EAFNOSUPPORT from Haiku.
I think fish tries to use IPV6 and something goes wrong. I'll look into the configure script, maybe there is a way to disable the IPV6 support...

**Comment by miqlas** [comment]
ibb536c

If i disable the IPv6 code, it works, so there is something definetly wrong with the IP v6 code. It coud be in the networking code in Haiku or in Python, because fish using Python for networking.
Can somebody help to nail it down? Thanks!

File: app-shells/fish/fish-2.2.0.recipe [hunk 1]
```diff
@@ -0,0 +1,69 @@
+SUMMARY="The friendly interactive shell"
+DESCRIPTION="
+Fish is a shell geared towards interactive use. Its features are \
+focused on user friendliness and discoverability. The language syntax \
+is simple but incompatible with other shell languages.
+"
```
Comment (waddlesplash) [inline app-shells/fish/fish-2.2.0.recipe]: Move `"` to previous line.

File: app-shells/fish/fish-2.2.0.recipe [hunk 2]
```diff
@@ -0,0 +1,69 @@
+SUMMARY="The friendly interactive shell"
+DESCRIPTION="
+Fish is a shell geared towards interactive use. Its features are \
```
Comment (waddlesplash) [inline app-shells/fish/fish-2.2.0.recipe]: Get rid of this newline.

File: app-shells/fish/fish-2.2.0.recipe [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	cmd:automake
+	cmd:libtoolize
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
+	cmd:which
+	cmd:sed
+	#cmd:doxygen >= 1.6.3
+	"
+
+PATCHES="fish-2.2.0.patch"
```
Comment (waddlesplash) [inline app-shells/fish/fish-2.2.0.recipe]: This goes beneath SOURCE_DIR.

**Comment by miqlas** [comment]
Is there anything else what i can do?

**Pull request merged**